ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HiveAugmentSnapshotMaterializationRule.java

```diff
@@ -125,11 +129,20 @@ public void onMatch(RelOptRuleCall call) {
     Table table = hiveTable.getHiveTableMD();
 
     SnapshotContext mvMetaTableSnapshot = mvMetaStoredSnapshot.get(table.getFullyQualifiedName());
-    if (mvMetaTableSnapshot.equals(table.getStorageHandler().getCurrentSnapshotContext(table))) {
+    if (table.getStorageHandler() == null) {
+      throw new UnsupportedOperationException(String.format("Table %s does not have Storage handler defined. " +
+          "Mixing native and non-native tables in a materialized view definition is currently not supported!",
+          table.getFullyQualifiedName()));
+    }
+    if (Objects.equals(mvMetaTableSnapshot, table.getStorageHandler().getCurrentSnapshotContext(table))) {
       return;
     }
 
-    table.setVersionIntervalFrom(Long.toString(mvMetaTableSnapshot.getSnapshotId()));
+    Long snapshotId = null;
+    if (mvMetaTableSnapshot != null) {
```

Comment:
````
Long snapshotId = mvMetaTableSnapshot != null ? mvMetaTableSnapshot.getSnapshotId() : null;
table.setVersionIntervalFrom(String.valueOf(snapshotId));
````

Reply:
`String.valueOf(null)` returns string literal `"null"`. This is not a valid snapshotId

Reply:
sorry, `Objects.toString(obj, null)`

Reply:
refactored